clients/drcachesim/tracer/output.cpp

```diff
@@ -1365,21 +1381,19 @@ exit_thread_io(void *drcontext)
     // Append a thread exit marker and output remaining records for this thread
     // if we are still in a tracing mode or the thread has data from a prior window
     // that it never wrote out.
-    if (is_in_tracing_mode(tracing_mode.load(std::memory_order_acquire)) ||
-        (has_tracing_windows() &&
+    if ((has_tracing_windows() &&
```

Comment:
It'd be nice to split this huge condition into multiple booleans, each with its own relevant comments atop it.

Reply:
Done.